setup.py

```diff
@@ -1,8 +1,53 @@
-from distutils.core import setup
-from Cython.Build import cythonize
-import numpy
-
-setup(
-    ext_modules=cythonize("*.pyx", language="c++"),
-    include_dirs=[numpy.get_include()]
-)
+from setuptools import Extension, find_packages, setup
+
+if __name__ == '__main__':
+  # see https://stackoverflow.com/questions/2379898/make-distutils-look-for-numpy-header-files-in-the-correct-place
+  # for the approach to pushing numpy an cython dependencies into extension building only
+  try:
+    # if Cython is available, we will rebuild from the pyx file directly
+    from Cython.Distutils import build_ext
+    sources = ['mdlp/_mdlp.pyx']
+  except:
+    # else we build from the cpp file included in the distribution
+    from setuptools.command.build_ext import build_ext
+    sources = ['mdlp/_mdlp.cpp']
+
+  class custom_build_ext(build_ext):
+    """
+    custom class to add numpy include dirs when we are building the extension.
```

Comment:
Please read the PEP 8 recommendations for docstrings here: https://www.python.org/dev/peps/pep-0257/#multi-line-docstrings

Reply:
great thank you.